src/main/java/de/eldoria/pickmeup/scheduler/TrailHandler.java

```diff
@@ -18,17 +18,20 @@ public TrailHandler(Plugin plugin) {
 
     @Override
     protected void execute(Entity entity) {
-        if (!entity.isValid()) {
+        if (entity == null || !entity.isValid() || entity.isDead()) {
```

Comment:
An entity is not valid if its dead. this is a duplicate check. An entity passed here will never be null, otherwise it couldnt be launched in the first palace
```suggestion
        if (!entity.isValid()) {
```